### PR TITLE
Implement solution suggested in issue #16 on multithreading

### LINKED ISSUE #16: Should fake time work outside main thread if specified explicitly?
Hi,

While debugging an [issue on a different project](https://github.com/pytest-dev/pytest-xdist/issues/56) I used `python-libfaketime` and accidentally noticed the `fake_time` context manager [has an argument](https://github.com/simon-weber/python-libfaketime/blob/master/libfaketime/__init__.py#L104) `only_main_thread=True`. Curious I tried `only_main_thread=False`, but got an exception:

```
self = <libfaketime.fake_time object at 0x107308c50>

    def _should_patch_uuid(self):
        return hasattr(uuid, '_uuid_generate_time') and \
>               not self._prev_spec and \
                self._should_fake()
E       AttributeError: 'fake_time' object has no attribute '_prev_spec'
```

At first I thought it is just simply the case of `_prev_spec` missing and changed it to `getattr(self, '_prev_spec', None)`, but now, upon closer inspection of the issue, it is [only being set](https://github.com/simon-weber/python-libfaketime/blob/master/libfaketime/__init__.py#L114) when `only_main_thread=True`. 

Therefore before submitting any pull request I wonder - is this expected behavior?

### PR DESCRIPTION
**Pull request by jppgks**
Closes #16

### CODE REVIEW

File: libfaketime/__init__.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def end_callback(instance):
 
 
 class fake_time(ContextDecorator):
-    def __init__(self, datetime_spec, only_main_thread=True):
+    def __init__(self, datetime_spec, only_main_thread=False):
```
Comment (simon-weber) [inline libfaketime/__init__.py]: Let's leave this as it was so the change is backwards-compatible. Callers who need it can just manually set it.

**Comment by jppgks** [comment]
@simon-weber: I left the default for `only_main_thread` as was.

File: libfaketime/__init__.py (hunk 1 above)
Comment (jppgks) [inline libfaketime/__init__.py]: @simon-weber: I edited the default for `only_main_thread` to as it was.

**Comment by simon-weber** [comment]
Awesome; thanks!

I'll plan to cut a release this weekend.

**Comment by jppgks** [comment]
Perfect!

**Comment by simon-weber** [comment]
Ok, 0.4.2 is up with these changes.

**Pull request merged**